src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java

```diff
@@ -119,8 +119,15 @@ public static boolean RequestHTTPS(
 
       if (fileStr.charAt(0) != '[' && fileStr.charAt(0) != '{') {
         // json has odd characters at the front, trim them to make valid json
-        int i = fileStr.indexOf("{");
-        fileStr = fileStr.substring(i);
+        int jsonStartIndex = fileStr.indexOf("{");
+
+        int notFoundIndex = -1;
+        if (jsonStartIndex == notFoundIndex) {
+          connected = true;
+          Logger.LOG_SERIOUS("Response received from OSIsoft server was not valid!");
+          return connected;
+        }
+        fileStr = fileStr.substring(jsonStartIndex);
       }
       JSONTokener JsonT = new JSONTokener(fileStr);

```

Comment:
Formatting is a bit weird here
```suggestion
        int notFoundIndex = -1;
        int jsonStartIndex = fileStr.indexOf("{");

        if (jsonStartIndex == notFoundIndex) {
          connected = true;
          Logger.LOG_SERIOUS("Response received from OSIsoft server was not valid!");
          return connected;
        }

        fileStr = fileStr.substring(jsonStartIndex);
      }

      JSONTokener JsonT = new JSONTokener(fileStr);

```